Description: Fix TLS Poodle
 Fix TLS Poodle so it checks the entire padding length.  
 This was caused by an off by one error.
 Patch created by Hanno Boeck (https://hboeck.de/).

---
Bug-Ubuntu: https://launchpad.net/bugs/1510163

--- gnutls26-2.12.23.orig/lib/gnutls_cipher.c
+++ gnutls26-2.12.23/lib/gnutls_cipher.c
@@ -569,7 +569,7 @@ _gnutls_ciphertext2compressed (gnutls_se
        * because there is a timing channel in that memory access (in certain CPUs).
        */
       if (_gnutls_version_has_variable_padding (ver) && pad_failed == 0)
-        for (i = 2; i <= pad; i++)
+        for (i = 2; i <= pad+1; i++)
           {
             if (ciphertext.data[ciphertext.size - i] != pad)
               pad_failed = GNUTLS_E_DECRYPTION_FAILED;
